server/app/views/components/Modal.java

```diff
@@ -97,6 +93,8 @@ public ModalBuilder setTriggerButtonText(String triggerButtonText) {
     }
 
     public ModalBuilder setTriggerButtonContent(Tag triggerButtonContent) {
+      Preconditions.checkState(
+          "button".equals(triggerButtonContent.getTagName()), "content must be of type button");
       this.triggerButtonContent = Optional.ofNullable(triggerButtonContent);
```

Comment:
note (non-blocking): the newer version of j2html has java types for each element which will eliminate the need for this sort of thing. working with Sami to get https://github.com/seattle-uat/civiform/pull/2281 merged would be a good fixit week item

Reply:
@bion I think I'm getting closed. pre recent merge there were just a couple types of browser tests failing. Anyhow, could you run the CI tests for me rn?

Reply:
Agreed that would be a huge improvement here. @syntapy, I'm happy to help (and / or review) if you run into any issues.

I'll be out 5/27-5/31, but happy to coordinate with you on Wednesday!

Reply:
@bion @clouser2 It seems like I can just remove that line then right? 96 & 97?

Reply:
@clouser2 sorry I didn't see your comment until just a min ago

Reply:
Yah, removing lines 96 and 97 should be good. If Preconditions isn't used elsewhere in the file, you might be able to remove the import as well.